platform/windows/detect.py

```diff
@@ -256,6 +256,8 @@ def build_def_file(target, source, env: "SConsEnvironment"):
     cmdbase = "dlltool -m " + arch_aliases[env["arch"]]
     if env["arch"] != "x86_32":
         cmdbase += " --no-leading-underscore"
+    else:
+        cmdbase += " -k"
```

Comment:
It makes sense to me to avoid a negative here and flip this around:

```py
    if env["arch"] == "x86_32":
        cmdbase += " -k"
    else:
        cmdbase += " --no-leading-underscore"
```